packages/playwright-core/src/client/browserContext.ts

```diff
@@ -100,6 +102,13 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
       if (page)
         page.emit(Events.Page.Console, consoleMessage);
     });
+    this._channel.on('pageError', ({ error, page }) => {
+      const pageObject = Page.fromNullable(page);
```

Comment:
Since it's non-nullable in the `protocol.yml`:

```suggestion
      const pageObject = Page.from(page);
```